gstaichi/platform/cuda/detect_cuda.cpp

```diff
@@ -8,7 +8,24 @@ namespace gstaichi {
 
 bool is_cuda_api_available() {
 #if defined(TI_WITH_CUDA)
-  return lang::CUDADriver::get_instance_without_context().detected();
+  try {
+    auto &instance = lang::CUDADriver::get_instance_without_context();
```

Comment:
this was mostly associated with getting the c api to not seg fault tbh. But still, maybe good to be able to check for cuda devices without causing a seg fault?

Reply:
This is nice I think.